src/data-workspace/category-combo-table-body/use-value-matrix.js

```diff
@@ -0,0 +1,46 @@
+import { getIn } from 'final-form'
+import { useMemo, useState, useEffect } from 'react'
+import { useForm } from 'react-final-form'
+import { useBlurredField } from '../../shared/index.js'
+
+const createValueMatrix = (dataElements, sortedCOCs, formState) =>
+    dataElements.map((de) =>
+        sortedCOCs.map((coc) => getIn(formState.values, `${de.id}.${coc.id}`))
+    )
+
+/**
+ * Generates matrix (2d-array) of the values in a table, defined by dataElements and
+ * the categoryOptionCombos.
+ * @param {*} dataElements dataElements in order as rendered in table, these are "rows"
+ * @param {*} sortedCOCs categoryOptionCombos in order as rendered in table, these are the "columns"
+ */
+export const useValueMatrix = (dataElements = [], sortedCOCs = []) => {
+    const form = useForm()
+    const blurredField = useBlurredField()
+    const [valueMatrix, setValueMatrix] = useState(
+        createValueMatrix(dataElements, sortedCOCs, form.getState())
+    )
+    const affectedFieldsLookup = useMemo(
+        () =>
+            new Set(
+                dataElements.flatMap((de) =>
+                    sortedCOCs.map((coc) => `${de.id}.${coc.id}`)
+                )
+            ),
+        [dataElements, sortedCOCs]
+    )
+    const containsAffectedField = useMemo(
```

Comment:
I don't think `useMemo` is needed here. `set.has` is O(1), and this is a primitive value, so using it directly should be fine and is probably more performant than `useMemo` would be.